app/controllers/application_controller.rb

```diff
@@ -20,6 +20,25 @@ class ApplicationController < ActionController::Base
 
   layout 'application'
 
+  # A common method for controllers to find a Dor object.  It accepts any valid
+  # PID (druid:aa111aa1111) or DRUID (aa111aa1111).  DruidTools::Druid parses
+  # input and returns a uniform PID, which is input to Dor.find to obtain a Dor
+  # object.
+  # @see https://github.com/sul-dlss/druid-tools
+  # @param druid [String]
+  # @return object [ActiveFedora::Base] any Dor object
+  # @raises ArgumentError
+  def find_druid(druid)
+    druid = DruidTools::Druid.new(druid).druid
+    Dor.find(druid)
+  rescue ArgumentError
+    flash[:error] = "Invalid DRUID: #{druid}"
+    render status: 400, text: flash[:error] and return
+  rescue ActiveFedora::ObjectNotFoundError # => e
+    flash[:error] = 'Object does not exist in Fedora.'
+    render status: 404, text: flash[:error] and return
```

Comment:
while i think hound's complaint is indeed off the mark, i'm actually concerned about the use of `flash` and `render` at all in this method.

`find_druid` gets called from `parse_object`, which gets called from `can_manage_object?`.  `can_manage_object?` is a method on the User model, and as such, it seems wrong for it to do controllerish things like rendering.  more practically, and less philosophically, `User#can_manage_object?` is the sort of thing you might want to call from the console, or a delayed job, possibly indirectly, and there's nothing to indicate that calling it outside the context of a web request is inappropriate.

it's true that it's mostly called directly from controllers, but maybe there should be a smaller method to do the druid tools thing, and a wrapper method that does error handling for use by the controllers that call `find_druid` directly.  though i don't think this extra level of in-line error handling is actually necessary either.  i think it'd be preferable to just let `ArgumentError` or `ObjectNotFoundError` bubble up and get handled by Rails.  maybe `ArgumentError` could get handled in application_controller in the same fashion that application_controller currently handles `ObjectNotFoundError`.  something like:
```ruby
rescue_from ArgumentError, with: -> { render text: 'Bad Request.  Please check the url and post data for syntax errors.  For item requests, ensure that a valid DRUID is specified.', status: 400 }
```